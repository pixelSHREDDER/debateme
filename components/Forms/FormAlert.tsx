import { Alert } from '@mantine/core'
import {
  IconHeartBroken,
  IconInfoCircle,
  IconThumbUp,
} from '@tabler/icons-react'
import { useId } from 'react'

const enum AlertType {
  Success,
  Warning,
  Error
}

interface IFormAlert {
  message: string,
  title: string,
  testid?: string,
  type: AlertType,
}

const colors = ['green', 'yellow', 'red']

export default function FormAlert(props: IFormAlert) {
  const id = useId()

  function getIcon() {
    if (props.type === AlertType.Success) {
      return <IconThumbUp />
    }

    if (props.type === AlertType.Warning) {
      return <IconInfoCircle />
    }

    return <IconHeartBroken />
  }

  function getMessage() {
    switch (props.message) {
      case 'Please fix the following errors: String must contain at least 1 character(s)':
        return 'Looks like your turn is empty!'
      default:
        return props.message
    }
  }

  return (
    <Alert
      data-testid={props.testid || ''}
      id={`alert_form_alert_${id}`}
      aria-live="polite"
      variant="light"
      color={colors[props.type]}
      radius="lg"
      title={props.title}
      my={20}
      icon={getIcon()}>
        {getMessage()}
    </Alert>
  )
}
