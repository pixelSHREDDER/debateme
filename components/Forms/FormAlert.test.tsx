import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import ProviderWrapper from '@/tests/helpers/ProviderWrapper'
import FormAlert from './FormAlert'

describe('Form alert component', () => {
  test('renders success alert correctly', () => {
    render(
      <FormAlert message="test success message" title="test success title" type={0} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByRole('alert')).toMatchSnapshot()
  })

  test('renders warning alert correctly', () => {
    render(
      <FormAlert message="test warning message" title="test warning title" type={1} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByRole('alert')).toMatchSnapshot()
  })

  test('renders error alert correctly', () => {
    render(
      <FormAlert message="test error message" title="test error title" type={2} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByRole('alert')).toMatchSnapshot()
  })

  test('Displays correct message for specific errors', () => {
    render(
      <FormAlert message="Please fix the following errors: String must contain at least 1 character(s)" title="" type={2} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByRole('alert')).toMatchSnapshot()
  })
})
