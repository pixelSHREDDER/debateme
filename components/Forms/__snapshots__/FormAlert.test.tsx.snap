// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form alert component > Displays correct message for specific errors 1`] = `
<div
  aria-describedby="alert_form_alert_:r9:-body"
  aria-live="polite"
  class="m_66836ed3 mantine-Alert-root"
  data-testid=""
  data-variant="light"
  id="alert_form_alert_:r9:"
  role="alert"
  style="--alert-radius: var(--mantine-radius-lg); --alert-bg: var(--mantine-color-red-light); --alert-color: var(--mantine-color-red-light-color); --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-block: calc(1.25rem * var(--mantine-scale));"
>
  <div
    class="m_a5d60502 mantine-Alert-wrapper"
  >
    <div
      class="m_667f2a6a mantine-Alert-icon"
    >
      <svg
        class="tabler-icon tabler-icon-heart-broken "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 12.572l-7.5 7.428l-7.5 -7.428a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
        />
        <path
          d="M12 6l-2 4l4 3l-2 4v3"
        />
      </svg>
    </div>
    <div
      class="m_667c2793 mantine-Alert-body"
    >
      <div
        class="m_7fa78076 mantine-Alert-message"
        data-variant="light"
        id="alert_form_alert_:r9:-body"
      >
        Looks like your turn is empty!
      </div>
    </div>
  </div>
</div>
`;

exports[`Form alert component > renders error alert correctly 1`] = `
<div
  aria-describedby="alert_form_alert_:r6:-body"
  aria-labelledby="alert_form_alert_:r6:-title"
  aria-live="polite"
  class="m_66836ed3 mantine-Alert-root"
  data-testid=""
  data-variant="light"
  id="alert_form_alert_:r6:"
  role="alert"
  style="--alert-radius: var(--mantine-radius-lg); --alert-bg: var(--mantine-color-red-light); --alert-color: var(--mantine-color-red-light-color); --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-block: calc(1.25rem * var(--mantine-scale));"
>
  <div
    class="m_a5d60502 mantine-Alert-wrapper"
  >
    <div
      class="m_667f2a6a mantine-Alert-icon"
    >
      <svg
        class="tabler-icon tabler-icon-heart-broken "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.5 12.572l-7.5 7.428l-7.5 -7.428a5 5 0 1 1 7.5 -6.566a5 5 0 1 1 7.5 6.572"
        />
        <path
          d="M12 6l-2 4l4 3l-2 4v3"
        />
      </svg>
    </div>
    <div
      class="m_667c2793 mantine-Alert-body"
    >
      <div
        class="m_6a03f287 mantine-Alert-title"
      >
        <span
          class="m_698f4f23 mantine-Alert-label"
          id="alert_form_alert_:r6:-title"
        >
          test error title
        </span>
      </div>
      <div
        class="m_7fa78076 mantine-Alert-message"
        data-variant="light"
        id="alert_form_alert_:r6:-body"
      >
        test error message
      </div>
    </div>
  </div>
</div>
`;

exports[`Form alert component > renders success alert correctly 1`] = `
<div
  aria-describedby="alert_form_alert_:r0:-body"
  aria-labelledby="alert_form_alert_:r0:-title"
  aria-live="polite"
  class="m_66836ed3 mantine-Alert-root"
  data-testid=""
  data-variant="light"
  id="alert_form_alert_:r0:"
  role="alert"
  style="--alert-radius: var(--mantine-radius-lg); --alert-bg: var(--mantine-color-green-light); --alert-color: var(--mantine-color-green-light-color); --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-block: calc(1.25rem * var(--mantine-scale));"
>
  <div
    class="m_a5d60502 mantine-Alert-wrapper"
  >
    <div
      class="m_667f2a6a mantine-Alert-icon"
    >
      <svg
        class="tabler-icon tabler-icon-thumb-up "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7 11v8a1 1 0 0 1 -1 1h-2a1 1 0 0 1 -1 -1v-7a1 1 0 0 1 1 -1h3a4 4 0 0 0 4 -4v-1a2 2 0 0 1 4 0v5h3a2 2 0 0 1 2 2l-1 5a2 3 0 0 1 -2 2h-7a3 3 0 0 1 -3 -3"
        />
      </svg>
    </div>
    <div
      class="m_667c2793 mantine-Alert-body"
    >
      <div
        class="m_6a03f287 mantine-Alert-title"
      >
        <span
          class="m_698f4f23 mantine-Alert-label"
          id="alert_form_alert_:r0:-title"
        >
          test success title
        </span>
      </div>
      <div
        class="m_7fa78076 mantine-Alert-message"
        data-variant="light"
        id="alert_form_alert_:r0:-body"
      >
        test success message
      </div>
    </div>
  </div>
</div>
`;

exports[`Form alert component > renders warning alert correctly 1`] = `
<div
  aria-describedby="alert_form_alert_:r3:-body"
  aria-labelledby="alert_form_alert_:r3:-title"
  aria-live="polite"
  class="m_66836ed3 mantine-Alert-root"
  data-testid=""
  data-variant="light"
  id="alert_form_alert_:r3:"
  role="alert"
  style="--alert-radius: var(--mantine-radius-lg); --alert-bg: var(--mantine-color-yellow-light); --alert-color: var(--mantine-color-yellow-light-color); --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-block: calc(1.25rem * var(--mantine-scale));"
>
  <div
    class="m_a5d60502 mantine-Alert-wrapper"
  >
    <div
      class="m_667f2a6a mantine-Alert-icon"
    >
      <svg
        class="tabler-icon tabler-icon-info-circle "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 12a9 9 0 1 0 18 0a9 9 0 0 0 -18 0"
        />
        <path
          d="M12 9h.01"
        />
        <path
          d="M11 12h1v4h1"
        />
      </svg>
    </div>
    <div
      class="m_667c2793 mantine-Alert-body"
    >
      <div
        class="m_6a03f287 mantine-Alert-title"
      >
        <span
          class="m_698f4f23 mantine-Alert-label"
          id="alert_form_alert_:r3:-title"
        >
          test warning title
        </span>
      </div>
      <div
        class="m_7fa78076 mantine-Alert-message"
        data-variant="light"
        id="alert_form_alert_:r3:-body"
      >
        test warning message
      </div>
    </div>
  </div>
</div>
`;
